addons/website_slides/static/src/js/portal_composer.js

```diff
@@ -0,0 +1,36 @@
+odoo.define('website_slides.portal.composer', function (require) {
+    'use strict';
+
+var portalComposer = require('portal.composer');
+var PortalComposer = portalComposer.PortalComposer;
+
+/**
+ * PortalComposer
+ *
+ * Extends to allow reviews of courses without writing message or add attachments.
+ */
+PortalComposer.include({
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onSubmitButtonClick: function (ev) {
+        ev.preventDefault();
+        var $websiteSlidesRatingComposer = ev.currentTarget.closest('.o_website_slides_rating_popup_composer');
+        // Special case : in slide.channel model, a strictly positive rating is submitted. We want to submit the
+        // comment and avoid the error message throwing when the message is empty / no attachments exist.
+        if ($websiteSlidesRatingComposer && this.$input.val() > 0) {
+            return this._chatterPostMessage(ev.currentTarget.getAttribute('data-action')).then((result) => {
+                const $modal = this.$el.closest('#ratingpopupcomposer');
+                $modal.on('hidden.bs.modal', () => {
+                  this.trigger_up('reload_rating_popup_composer', result);
+                });
+                $modal.modal('hide');
+            });
+        } else {
+            return this._super(...arguments);
+        }
```

Comment:
I would do something like this. What do you think ?

>         // Special case : in slide.channel model, a strictly positive rating is submitted. We want to submit the
>         // comment and avoid the error message throwing when the message is empty / no attachments exist.
>         if (ev.currentTarget.closest('.o_website_slides_rating_popup_composer')) {
>             if (parseFloat(this.$input.val()) === 0) {
>                 const error = _t('The rating is required. Please make sure to select one before sending your review.');
>                 this.$(".o_portal_chatter_composer_error").text(error).removeClass('d-none');
>                 return Promise.reject();
>             }
>             return this._chatterPostMessage(ev.currentTarget.getAttribute('data-action')).then((result) => {
>                 const $modal = this.$el.closest('#ratingpopupcomposer');
>                 $modal.on('hidden.bs.modal', () => {
>                   this.trigger_up('reload_rating_popup_composer', result);
>                 });
>                 $modal.modal('hide');
>             });
>         }
>         return this._super(...arguments);

Reply:
Hey @jeh-odoo, Thanks for review, I will do the changes, just a remark on this comment :)

Atm, one user can leave a comment without rating, actually. If he does not select any stars, a review can still be posted a long as there is a message / an attachment. In testing, it only is possible when it is the first review of a userThat is why I've added this positivity constraint : 
1. If positive, the rating exists and will be taken into account (in the mean score, ...), so we allow no comment.
2. if zero, it is not set. In that case we call super as the constraint of having either a message / attachment make sense again. Selecting no stars will not update any score. So I left the super do the error message work :p

What do you think?

Reply:
Yeah, I've talked to Luc about this last week. But it's a little odd to have two different cases where one you can send a review with 0 star with a comment and one where you need to select a rating before sending it without a comment.
From my point of view there are multiple possibilities:

- We can send a review without any constraint
- We set a constraint on the rating for it to be always set in elearning (no more 0 star allowed)
- We keep this that way (because 0 star needs an explanation)

I let you decide what you prefer. Also I don't know if letting a review with 0 star was intented in the first place because if you try to modify your review you can't select 0 star anymore.

Reply:
Yes indeed. Also, side note : it feels weird that one user can post as many reviews as he wants and modify the score this way, I don't know why this is done this way. I think we can go for the error message you proposed, I've never seen a mix of comments and rated reviews somewhere else as far as I remember.

Reply:
For the multi review thing it's a bug that we are aware of. There is a task somewhere in the pipe about it :)